corehq/apps/users/static/users/js/mobile_workers.ng.js

```diff
@@ -0,0 +1,252 @@
+        'ngResource',
+        'ngRoute',
+        'ng.django.rmi',
+        'ngMessages'
+    ]);
+
+    var $formElements = {
+        username: function () {
+            return $('#id_username').parent();
+        }
+    };
+
+    var visualFormCtrl = {
+        usernameClear: function () {
+            $formElements.username()
+                .removeClass('has-error has-pending has-success');
+        },
+        usernameSuccess: function () {
+            $formElements.username()
+                .removeClass('has-error has-pending')
+                .addClass('has-success');
+        },
+        usernamePending: function () {
+            $formElements.username()
+                .removeClass('has-error has-success')
+                .addClass('has-pending');
+        },
+        usernameError: function () {
+            $formElements.username()
+                .removeClass('has-success has-pending')
+                .addClass('has-error');
+        }
+    };
+
+    var STATUS = {
+        NEW: 'new',
+        PENDING: 'pending',
+        WARNING: 'warning',
+        SUCCESS: 'success',
+        RETRIED: 'retried'
+    };
+
+    var USERNAME_STATUS = {
+        PENDING: 'pending',
+        TAKEN: 'taken',
+        AVAILABLE: 'available',
+        ERROR: 'error'
+    };
+
+    mobileWorkers.constant('customFields', []);
+    mobileWorkers.constant('customFieldNames', []);
+
+    var MobileWorker = function (data) {
+        var self = this;
+        self.creationStatus = STATUS.NEW;
+
+
+        self.username = data.username || '';
+        self.first_name = data.first_name || '';
+        self.last_name = data.last_name || '';
+        self.editUrl = data.editUrl || '';
+
+        self.password = '';
+
+        self.customFields = {};
+
+        self.isPending = function () {
+            return self.creationStatus === STATUS.PENDING;
+        };
+
+        if (_.isArray(data.customFields)) {
+            _.each(data.customFields, function (key) {
+                self.customFields[key] = '';
+            });
+        } else if (_.isObject(data.customFields)) {
+            self.customFields = data.customFields;
+        }
+    };
+
+    var mobileWorkerControllers = {};
+
+    mobileWorkerControllers.MobileWorkerCreationController = function (
+            $scope, workerCreationFactory, djangoRMI, customFields,
+            customFieldNames
+    ) {
+        $scope._ = _;  // make underscore available
+        $scope.mobileWorker = {};
+        $scope.usernameAvailabilityStatus = null;
+        $scope.usernameStatusMessage = null;
+        $scope.workers = [];
+        $scope.customFormFields = customFields;
+        $scope.customFormFieldNames = customFieldNames;
+
+        $scope.initializeMobileWorker = function (mobileWorker) {
+            visualFormCtrl.usernameClear();
+            $scope.usernameAvailabilityStatus = null;
+            $scope.usernameStatusMessage = null;
+
+            if (!!mobileWorker) {
```

Comment:
nit: don't think you actually need the `!!`. also when is this ever false? `{}` evaluates to true which seems to be the default